src/hooks/__tests__/useCredentialApi.test.ts

```diff
@@ -0,0 +1,140 @@
+};
+
+const mockCredential2: CredentialType = {
+  id: 2,
+  name: 'Test Credential',
+  created_at: new Date(),
+  updated_at: new Date(),
+  cred_type: 'Test Type',
+  username: 'testuser',
+  password: 'testpassword',
+  ssh_keyfile: 'testkeyfile',
+  auth_token: 'testtoken',
+  ssh_passphrase: 'testpassphrase',
+  become_method: 'testmethod',
+  become_user: 'testuser',
+  become_password: 'testpassword',
+  sources: [mockSource]
+};
+
+jest.mock('axios');
+
+describe('useDeleteCredentialApi', () => {
+  let mockOnAddAlert: jest.Mock;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  let result: any;
+
+  beforeEach(() => {
+    mockOnAddAlert = jest.fn();
+    const hook = renderHook(() => useDeleteCredentialApi(mockOnAddAlert));
+    result = hook.result;
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('successfully deletes credentials (no attached sources)', async () => {
+    (axios.post as jest.MockedFunction<typeof axios.post>).mockResolvedValue({
+      data: { skipped: [], deleted: ['1'] }
+    });
+
+    await act(async () => {
+      await result.current.deleteCredentials(mockCredential1);
+    });
+
+    expect(axios.post).toHaveBeenCalledWith(
+      `${process.env.REACT_APP_CREDENTIALS_SERVICE_BULK_DELETE}`,
+      { ids: [mockCredential1.id] }
+    );
+    expect(mockOnAddAlert).toHaveBeenCalledWith(expect.objectContaining({ variant: 'success' }));
+  });
+
+  it('shows warning alert (some deleted, some skipped)', async () => {
+    (axios.post as jest.MockedFunction<typeof axios.post>).mockResolvedValue({
+      data: { skipped: [mockCredential2.id], deleted: [mockCredential1.id] }
+    });
+
+    await act(async () => {
+      await result.current.deleteCredentials([mockCredential2, mockCredential1]);
+    });
+
+    expect(mockOnAddAlert).toHaveBeenCalledWith(expect.objectContaining({ variant: 'warning' }));
+  });
+
+  it('shows error alert (all skipped)', async () => {
+    (axios.post as jest.MockedFunction<typeof axios.post>).mockResolvedValue({
+      data: { skipped: [{ credential: mockCredential2.id }], deleted: [] }
+    });
+
+    await act(async () => {
+      await result.current.deleteCredentials(mockCredential2);
+    });
+
+    expect(mockOnAddAlert).toHaveBeenCalledWith(expect.objectContaining({ variant: 'danger' }));
+  });
+
+  it('should handle Axios errors gracefully', async () => {
+    const error = new AxiosError('Request failed with status code 400');
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    (error as any).response = {
+      status: 400,
+      data: { detail: 'Request failed with status code 400' }
+    };
+    (axios.post as jest.MockedFunction<typeof axios.post>).mockRejectedValue(error);
+
+    await act(async () => {
+      await result.current.deleteCredentials(mockCredential1);
+    });
+
+    expect(mockOnAddAlert).toHaveBeenCalledWith(expect.objectContaining({ variant: 'danger' }));
+  });
+});
```

Comment:
I compliment the thoroughness of these tests, however... we shouldn't be testing axios. Axios is already tested we should be testing what we're actually doing, submitting formatted data and displaying alerts

required: 
- I've opened up a PR that will provide you with some enhancements for testing, **bring that commit into this work**... we'll be merging that work before yours since it helps with your other PRs too
- let's focus your unit tests and try to make sure we house our mocks under the `describe`
   - we shouldn't actually need full fledge mock responses... 
- I've removed the detailed mocks and opted for simplistic data. The mocks look like they took time, apologies for removing them. The reason, we're not necessarily concerned about e2e level detail at this point is that these are unit tests and we are using Typescript. Typescript should help us catch issues at a detail level if our typings are setup correctly. We wrote functions that don't really care about anything but their piece, so from that perspective we just need enough to be to test that functionality and trust Typescript. 
- I've broken the tests out into something that uses a more simplistic axios mock using the `jest.spyOn` method where we'll detect the `calls` to with snapshots... we can apply this `calls` concept to the `onAddAlert` callback too

- And I've added direct tests for the success and error callbacks now that we're exporting them, and we're no longer bogged down trying to mock out the entire hook

```
describe('useDeleteCredentialApi', () => {
  let mockOnAddAlert;
  let hookResult;

  beforeEach(() => {
    mockOnAddAlert = jest.fn();
    const hook = renderHook(() => useDeleteCredentialApi(mockOnAddAlert));
    hookResult = hook?.result?.current;
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  // Let's keep the axios call simple, we can use a jest.spyOn instead, ignore the typings on spyOn they also appear out-of-date/busted similar to `toMatchSnapshot`
  // we're going to use `mock.calls` and pull out the "attempt" at an API response
  it('should attempt an api call to delete credentials', () => {
    const { apiCall } = hookResult;
    const mockAxios = jest.fn();
    jest.spyOn(axios, 'post').mockImplementation(mockAxios);

    apiCall([456, 789]);
    expect(mockAxios.mock.calls).toMatchSnapshot('apiCall');
  });

  it('should attempt to delete single, and multiple credentials', () => {
    const { deleteCredentials } = hookResult;
    const mockAxios = jest.fn();
    jest.spyOn(axios, 'post').mockImplementation(mockAxios);

    deleteCredentials({ id: 123 });
    deleteCredentials([{ id: 456 }, { id: 789 }]);
    expect(mockAxios.mock.calls).toMatchSnapshot('deleteCredentials');
  });

  it('should process an API success response', () => {
    const { callbackSuccess } = hookResult;

    // danger
    callbackSuccess(
      {
        data: {
          skipped: [{ credential: 123 }, { credential: 456 }]
        }
      },
      [
        { id: 123, name: 'Lorem' },
        { id: 456, name: 'Ipsum' }
      ]
    );

    // warning
    callbackSuccess(
      {
        data: {
          deleted: [456],
          skipped: [{ credential: 123 }]
        }
      },
      [
        { id: 123, name: 'Lorem' },
        { id: 456, name: 'Ipsum' }
      ]
    );

    // success
    callbackSuccess(
      {
        data: {
          deleted: [123, 456]
        }
      },
      [
        { id: 123, name: 'Lorem' },
        { id: 456, name: 'Ipsum' }
      ]
    );

    expect(mockOnAddAlert.mock.calls).toMatchSnapshot('callbackSuccess');
  });

  it('should process an API error response', () => {
    const { callbackError } = hookResult;

    callbackError(
      {
        response: {
          data: {
            detail: 'Lorem ipsum'
          }
        }
      },
      [
        { id: 123, name: 'Lorem' },
        { id: 456, name: 'Ipsum' }
      ]
    );

    callbackError(
      {
        response: {
          data: {
            message: 'Dolor sit'
          }
        }
      },
      [
        { id: 123, name: 'Lorem' },
        { id: 456, name: 'Ipsum' }
      ]
    );

    callbackError(
      {
        message: 'Amet'
      },
      [
        { id: 123, name: 'Lorem' },
        { id: 456, name: 'Ipsum' }
      ]
    );

    callbackError({}, [
      { id: 123, name: 'Lorem' },
      { id: 456, name: 'Ipsum' }
    ]);

    expect(mockOnAddAlert.mock.calls).toMatchSnapshot('callbackError');
  });
});
```